PayPal/src/main/java/com/braintreepayments/api/PayPalClient.java

```diff
@@ -54,91 +77,97 @@ private static Exception createBrowserSwitchError() {
     }
 
     /**
-     * Starts the One-Time Payment flow for PayPal.
+     * Tokenize a PayPal account for vault or checkout.
      *
      * @param activity Android FragmentActivity
-     * @param payPalRequest a {@link PayPalRequest} used to customize the request. An amount MUST be specified.
+     * @param payPalRequest a {@link PayPalRequest} used to customize the request.
      * @param callback {@link PayPalFlowStartedCallback}
      */
-    public void requestOneTimePayment(final FragmentActivity activity, final PayPalRequest payPalRequest, final PayPalFlowStartedCallback callback) {
-        if (payPalRequest.getAmount() != null) {
-            braintreeClient.sendAnalyticsEvent("paypal.single-payment.selected");
-            if (payPalRequest.shouldOfferCredit()) {
-                braintreeClient.sendAnalyticsEvent("paypal.single-payment.credit.offered");
-            }
-            
-            if (payPalRequest.shouldOfferPayLater()) {
-                braintreeClient.sendAnalyticsEvent("paypal.single-payment.paylater.offered");
-            }
+    public void tokenizePayPalAccount(final FragmentActivity activity, final PayPalRequest payPalRequest, final PayPalFlowStartedCallback callback) {
+       if (payPalRequest instanceof PayPalCheckoutRequest) {
+           requestOneTimePayment(activity, (PayPalCheckoutRequest) payPalRequest, callback);
+       } else if (payPalRequest instanceof PayPalVaultRequest) {
+           requestBillingAgreement(activity, (PayPalVaultRequest) payPalRequest, callback);
+       }
```

Comment:
If possible, I think we should invert which method is calling the other here. Since `requestOneTimePayment` and `requestBillingAgreement` are deprecated, we're going to delete the at some point in the future. It'll be easier to do that if the deprecated methods call `tokenizePayPalAccount` than if it's the other way around.

Reply:
We updated these deprecated methods to call the `tokenizePayPalAccount` method and added private methods for checkout vs vault flow.